### DISCUSSION TITLE
zulu scsi hd file storage on mpc 2000xl

### DISCUSSION THREAD

**Discussion by dubsetter** [ZuluSCSI/ZuluSCSI-firmware#362] [Q&A] [answered]
hey,
i created two hd. files on the zulu rp 2040 for the mpc
(other ids are cd iso files)
the pc shows 1, 8 gb per hd,
but in the mpc  it only shows 640 mb of storage.
i also tried various hd endings (hda, iso, img ) with no result.
im on os. 1.14 which according to  the net is capable of reading disks over 2 gb.
any idea?
sd card is a 32 gb sandisk
thanx...

**Comment by chickeneps** [comment]
Remember the MPC separates a drive/image into equally-sized letter-named "partitions". (Especially 1.14, which you should stick to using.) Of course it depends when the drive/image gets formatted, you choose on the MPC A-A or A-B or A-G etc. or A-Z. Since we are almost always dealing with large drives/images these days, you should always format as A-Z, so its divided into 26 partitions.

So, perhaps when you say "but in the mpc it only shows 640 mb of storage" it's just referring to one of the partitions (like Partition A) and it has 3 partitions... so 640 x 3 = 1.8GB like you said.

Is that it?

**Comment by dubsetter** [comment]
sorry, but when the mpc sees the hd file/disk (in the menue/disk management),
it says 640 mb not partitioned /partition A-A.
if u format it into 4 partitions (A-D) its 140 mb each.
so the mpc seems like to see the hd file as a 640 mb disk,
not 1,8 gb.
??

**Reply by chickeneps** [reply]
140 x 4 is 560, not 640. But anyways...

OK, so...

1) When you look at that image file on a computer, and look at the size, the computer reports that it's 1.8GB?
2) But when you look at the image file (as a drive, obviously) in the MPC, it essentially - through expirimention with formatting the partitions - says it's only 560mb/640mb?

If so, have you ever formatted the SD card you are using yourself? Or is it just like it was purchased form the store? When you look at the SD card on the computer, what format does it say it is?

Lastly, please submit the zululog.txt to the thread here.

**Comment by dubsetter** [comment]
sorry, i meant 160 mb each (4 partitions)
:)

the card came from the shop i bought the zulu card from with
two hd disk files and 3 cd iso files.
i did nt format the sd card by myself.
on the computer, the hd files are 1,8 gb big.
i can post the ini file tomorrow, as the card is in the studio right now.
same goes for the  sd card. (format)
thanx

**Comment by chickeneps** [comment]
Post the zuluscsi.ini and zululog.txt, and zip up the image file in question and post it here.

**Comment by dubsetter** [comment]
allright, thanx so far....
i know where to find the ini.file,
but wheres the log.text ?
also the zipped hd file will be quite big....
is that okay , to post it here?

**Comment by chickeneps** [comment]
zuluscsi.ini is a file you put on yourself. zululog.txt is always written by the ZuluSCSI unit itself, and is rewritten on every reboot.

(There is also a zuluerr.txt that is written when the firmware crashes, but that is of no concern at the moment.)

If you've powered the ZuluSCSI and no zululog.txt is written, then you have a more serious issue in that your SD card is refusing to be written to.

**Comment by dubsetter** [comment]
okay, i will have a look tomorrow at the card, 
might be i misunderstood/messed up the names (.log or .ini)
as there was a file on the card , when i looked at it this afternoon.
(the card can definetely be written to, as i already transfered files /programs from the mpc on hd.0
on the zulu.
is there a difference between naming the hd files iso or hda..(when creating them on the pc)
as the mpc recognized both of them as hd.disks (scsi-0, scsi-1...) ?

**Comment by dubsetter** [comment]
hey,
heres the log file,
i dont have any ini file on the card.
the card is formatted in fat32.
also attached is the HD1.file
(which is empty at the moment)
the HD0.file is already filled a bit with programs and samples form the mpc.
(though i guess its the same storage: 640 mb)
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/13830869/zululog.txt)
[HD1.zip](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/13830871/HD1.zip)

**Comment by chickeneps** [comment, marked as answer]
OK, looks like the problem is that you have too much junk on your SD card and what you THINK is ID 1 is not ID 1. From your log...

[987ms] -- Opening /Real guitars.iso for id:0 lun:0
[1092ms] ---- Configuring as removable drive based on image name
[1093ms] -- Opening /renaissance.iso for id:1 lun:0
[1159ms] ---- Configuring as removable drive based on image name
[1163ms] -- Ignoring /HD0.ISO, SCSI ID 0 is already in use!
[1163ms] -- Ignoring /HD1.img, SCSI ID 1 is already in use!

From this log, it seems that there's a undocumented feature in the firmware where - like "HD" and "CD" - an "RE" means the image is a removable drive and is added to the SCS ID list as such. I think you thought "Real guitars" and "renaissance" would be ignored but their first two characters as "re" instructed ZuluSCSI to see them as removable images/drives.

Make a "junk" folder on the SD card and put all the unused images in there as to hide them from ZuluSCSI.

**Comment by dubsetter** [comment]
ah okay,
so only the CD .iso files and the HD files should be on the card..
the rest in a junk folder on the card?

i already had transfered files from the mpc to "scsi-0"
and created subfolders on it.
does that mean i have to do this again,
when i put the mentioned files in the junk folder?

**Comment by chickeneps** [comment]
(First, never use the "CD" prefix, it's not necessary and just confusing for you.)

You're making this too complicated. Just make sure that everything in the root doesn't conflict as far as SCSI ID's are concerned. The SCSI ID for a image file is determined by the NUMBER that occurs right after the HD prefix.

The reason you had the problem you came here with is because you had a 640mb image file occupying ID1, but you thought it was another 1.8GB file occupying that ID. A case of a "imposter image file" =)

**Comment by dubsetter** [comment]
sorry, i did nt configure that card,
it came preinstalled from the shop.
(also that cd.iso files)
how should i know, that the firmware sees real guitars as an hd (scsi) disk?
im also no programmer or know what  a log .file  is saying or means.
thanx for the infos, 
as in other forums (mpc related ) nobody could  help me.

you said frist, that the first letters (RE) were causing this,
now you write about scsi numbers (HD.0,HD.1)
which only exist as i created two HD files on my own on the card.

an specific answer to my questions would be nice:

i already had transfered files from the mpc to "scsi-0"
and created subfolders on it.
does that mean i have to do this again,
when i put the mentioned files in the junk folder?

**Comment by chickeneps** [comment]
There's A LOT of misinformation out there. The Internet's great, but it also reveals how unintentionally dishonest people can be. Mostly where people give wrong advice, trying to be helpful, then never bother to erase their wrong advice - if they can ever erase it.

I didn't answer the question because I didn't understand it. If you did write to another image unintentionally, just rename that unintended image to an unused SCSI ID and access it on the MPC that way.

**Comment by dubsetter** [comment]
okay , thanx a lot  so far..
i wish i did configure that card by myself,
not that shop that sold the zulu card.

EDIT:
i was able to open" real guitars" with 7 zip,
and this is my scsi-0 (transfered files) on the mpc...:)

so i will put all files into junk (except that cd files and my two hd.files,
but before i will transfer the "real guitar" stuff onto the "new" HD.1 then.



greetz ,
lars

**Comment by chickeneps** [comment]
There's no "unmounting" of image files. The concept is simpler than you are thinking it is. At the most basic, my recommendation as far as MPC use goes:

1) In the root of the SD card, only have images files that start with "HD" and have unique numbers after the "HD", and do not include 6 since that is the self-ID of the MPC2000XL. Use ".IMG" as the extension. You are free to add a space (very important) after the ID number and any text you feel is important. Example: ID2 Super Beats.IMG

2) When you start up the MPC and you got SHIFT-DISK and scroll through the ID's, those are "mounted" until the next power-cycle of the MPC. There is no unmounting of anything, it's permanent for that session.

3) If you want to change images, etc. a) turn off the MPC b) put the SD card in your computer and make the wanted changes c) put the SD card back into your MPC, turn it on, and go for it.

**Comment by dubsetter** [comment]
yes,
i know that,
i only wanted to see if my files were now in real guitars or renaissance.
nevermind i could open the iso with 7zip.

thanx again, and will post tomorrow if i sucessfully create an 1,8 gb disk...:)

**Comment by chickeneps** [comment]
And, BTW, if you want to know a good program on the computer (Mac or Windows) that can look INSIDE these image files and see the contents and even edit them, look at Chicken Systems Translator [Free]. (I am the project lead and main writer of that program.) Sure, you can open the image using Mac (native) or Windows using a third-part mounter, or 7zip, but you won't see the partitions past A and they don't support the long-file-name scheme. Translator [Free] does. The difference between Free and paid is that paid allows conversion of other formats like Battery or SoundFont or anything into MPC200XL format.

**Comment by dubsetter** [comment]
ah good to know...
i tried free tanslator the other day, but i have no internet connection in the studio,
which cause that i have to fill in my code all the time.

**Comment by dubsetter** [comment]
i was in the studio today,
transfered my files onto the created Hd1 disk
(i deleted "renaissance " iso before)
scsi disk showed the whole 1,8 gb in 3 partitions.

then i deleted "real guitars" iso,
made a new HD0 with the same 1,8 gb ..
et voila...

im happy now...
:)

**Comment by chickeneps** [comment]
Good deal!
